PasswordManager/src/passwordmanager/PasswordClient.java

```diff
@@ -1,5 +1,107 @@
 package passwordmanager;
 
+import java.io.IOException;
+import java.net.Socket;
+
+import passwordmanager.communication.CommunicationProtocol;
+import passwordmanager.communication.CommunicationProtocol.CommunicationOperation;
+import passwordmanager.communication.CommunicationProtocol.ProtocolMode;
+import passwordmanager.communication.Query;
+import passwordmanager.communication.Response;
+import passwordmanager.config.Configuration;
+
+
+
 public class PasswordClient {
+	
+/**	
+ * 
+ *  
+ * @author Yemeri 
+ *  
+ */
+	
+/** Public classes because the PasswordController has to be able to call on them **/
+	
+	private CommunicationProtocol Protocol; 
+	
+	public PasswordClient(Configuration config) throws IOException {
+		Socket socket = new Socket(config.serverIp, config.serverPort);
+		Protocol = new CommunicationProtocol(socket, ProtocolMode.Client);
+	
+	}
+	
+	
+	
+	
+/**	Code below regards Credentials. Manipulates user passwords (Store/Retrieve/Modify/Delete) using PasswordServer, UserAcc, PasswordDatabase & Credentials. For passwords stored inside the Password manager **/
+	
+	
+	public Credential[] getCredential (UserAccount account) {
+		Query<UserAccount> query = new Query<UserAccount>("", CommunicationOperation.GetAllCredentials, account);
+		Response<Credential[]> response = Protocol.sendAndReceive(query);
+		return response.getData();
+		
+	} 
```

Comment:
To make it easier for the developers that are using this class, this method should probably be named "getCredentials" to emphasize the fact that multiple credentials are being fetched from the server.
```suggestion
	public Credential[] getCredentials (UserAccount account) {
		Query<UserAccount> query = new Query<UserAccount>("", CommunicationOperation.GetAllCredentials, account);
		Response<Credential[]> response = Protocol.sendAndReceive(query);
		return response.getData();
		
	} 
```